### ISSUE TITLE
None of the ADBAPI tests actually run

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8139]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8139|
|Type|enhancement|
|Created|2015-12-05 04:37:14Z|
|Branch|https://github.com/twisted/twisted/tree/adbapitests-8139|

We should probably have their dependencies and run at least a small portion of them.

<details><summary>Searchable metadata</summary>

```
trac-id__8139 8139
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_adbapitests_8139 branches/adbapitests-8139
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1449290234919305 1449290234919305
changetime__1449381990731832 1449381990731832
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [46419]) Branching to adbapitests-8139.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

I'm happy with this. It runs the tests, and cleans up the code a bit.

Please review, builders spun. The Ubuntu 14.04 is failing as it doesn't have the SQLite headers, I'm working on getting them on the builder.

Compare the coverage here: https://codecov.io/github/twisted/twisted?branch=adbapitests-8139 vs https://codecov.io/github/twisted/twisted?branch=trunk

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Great work! Congrats for the coverage improvements for the "enterprise" code!

---------

I see that firebird, postresql and mysql tests are not execute.

Maybe we should create a ticket for them so that people will know that those connectors are not tested / not supported... and also update the narrative docs to include this note.

------

Feel free to merge after Ubuntu 14.04 is fixed.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46438]) Merge adbapitests-8139: Make the adbapi tests actually run

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8139](https://github.com/twisted/twisted/issues/8139)
